src/server/index.ts

```diff
@@ -18,11 +27,30 @@ app.get('*', (_, res) => {
     res.sendFile(path.join(__dirname, 'homepage.html'));
 });
 
-io.on('connection', (socket) => {
-    socket.on('chatMessage', (msg: string) => {
-        io.emit('chatMessage', msg);
-    });
-});
+io.on(
+    'connection',
+    (socket: Socket<ClientToServerEvents, ServerToClientEvents>) => {
+        socket.on('chatMessage', (msg: string) => {
+            io.emit('chatMessage', msg);
+        });
+
+        socket.on('chooseName', (name: string) => {
+            if (!name) {
+                clearName(socket.id);
+                socket.emit('confirmName', '');
+                return;
+            }
+            if (!namesToIds.has(name)) {
+                namesToIds.set(name, socket.id);
+                socket.emit('confirmName', name);
+            }
```

Comment:
it fails if the name is already taken!

For the MVP we won't be adding any error messaging, but it's a worthy consideration